### PR TITLE
81359  Update conditional displays based on file uploads - form 10-7959c

### PR DESCRIPTION
**Pull request by michaelclement**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [X] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

## Summary

This PR improves the customizability of the `MissingFileOverview` component with new custom wording options for the confirmation page, and fixes a bug in the same that would sometimes display "required" file upload prompts from pages that were no longer conditionally present in the form. 

The latter issue happened when users answered questions that caused file upload pages to be required, but then went backwards in the form and changed their answers to no longer require those upload pages. The issue was that the `MissingFileOverview` page was not checking against the list of conditionally required pages, but instead it was just comparing against the overall list of pages and so it showed file uploads from hidden pages.

- Updates customizability of `mailInfo` text in `MissingFileOverview` on confirmation page (adds yellow incomplete banner)
- Updates copy on confirmation page for CHAMPVA OHI form (10-7959c)
- Fixes bug that would show ghost file uploads on the `MissingFileOverview` page
- Team: IVC CHAMPVA
- Flipper: NA

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/81359

## Testing done

- Manual
- Verified unit and E2E tests pass

### Issue 1 (confirmation page):
||Before|After|
|-|-|-|
|Confirmation Page|Screenshot 2024-07-29 at 14 21 53|Screenshot 2024-07-29 at 13 13 37|

### Issue 2: ghost file uploads on review
|         | Before (ghost uploads display) | After |
| ------- | ------ | ----- |
| Missing File Overview |before|after|

## What areas of the site does it impact?

IVC CHAMPVA forms that use `MissingFileOverview` (currently: 10-10d and 10-7959c).

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [X] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [X] Browser console contains no warnings or errors.
- [X] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

NA

### CODE REVIEW

File: src/applications/ivc-champva/10-7959C/containers/ConfirmationPage.jsx [hunk 1]
```diff
@@ -25,10 +28,15 @@ const heading = (
...
+      <h2>
+        You submitted your CHAMPVA Other Health Insurance Certification form
+        without required documents
+      </h2>
+    </VaAlert>
     <p>
       You’ll still need to send us these required documents in order for us to
-      process this application:
+      process this form:
     </p>
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-7959C/containers/ConfirmationPage.jsx:40]: Updates this to show a yellow warning when the user has not submitted all documents alongside their form.

File: src/applications/ivc-champva/10-7959C/containers/ConfirmationPage.jsx [hunk 2]
```diff
@@ -54,9 +69,12 @@ export function ConfirmationPage(props) {
     showMail: true,
     allPages: form.pages,
     fileNameMap: prefixFileNames(data, requiredFiles),
+    optionalDescription: '',
+    requiredDescription: '',
     requiredFiles,
     nonListNameKey: 'applicantName',
     mailingAddress: officeAddress,
+    mailPreamble,
     officeName: office,
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-7959C/containers/ConfirmationPage.jsx:78]: Adds optional new string parameters so we can customize the language in more places on the shared component.

File: src/applications/ivc-champva/shared/components/fileUploads/MissingFileOverview.jsx [hunk 3]
```diff
@@ -258,16 +263,13 @@ export default function MissingFileOverview({
...
-      pages,
+      sponsorConditionalPages,
       data,
-      verifier.identifyMissingUploads(
-        getConditionalPages(pages, data),
-        data,
-        true,
-      ),
+      verifier.identifyMissingUploads(sponsorConditionalPages, data, true),
     ).missingUploads,
```
Comment (michaelclement) [inline src/applications/ivc-champva/shared/components/fileUploads/MissingFileOverview.jsx:273]: This was where the bug was that showed ghost files on the `MissingFileOverview` component. Originally we were checking all `pages` in the form to find file uploads. Now we're only concerned with pages that are currently in the form flow (conditionally rendered by `depends` funcs)

File: src/applications/ivc-champva/10-7959C/containers/ConfirmationPage.jsx [hunk 4]
```diff
@@ -81,24 +99,15 @@ export function ConfirmationPage(props) {
...
         </h3>
-        {data.statementOfTruthSignature && (
+        {(data.statementOfTruthSignature || data.signature) && (
           <span className="veterans-full-name">
             <strong>Who submitted this form</strong>
             <br />
-            {data.statementOfTruthSignature}
-            <br />
+            {data.statementOfTruthSignature || data.signature}
           </span>
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-7959C/containers/ConfirmationPage.jsx:111]: Previously this was not accounting for keyname on custom signature field for this form.

File: src/applications/ivc-champva/10-7959C/containers/ConfirmationPage.jsx [hunk 5]
```diff
@@ -129,9 +138,12 @@ export function ConfirmationPage(props) {
...
       </div>
-      <a className="vads-c-action-link--green" href="https://www.va.gov/">
-        Go back to VA.gov
-      </a>
+      <h2 className="vads-u-font-size--h3">What to expect next</h2>
+      <p>
+        We’ll contact the number you listed on this form by mail or phone if we
+        have questions or need more information.
+      </p>
+      <VaLinkAction href="/" text="Go back to VA.gov" />
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-7959C/containers/ConfirmationPage.jsx:146]: Updates position of some copy and replaces link with newer `VaLinkAction` component.

File: src/applications/ivc-champva/shared/components/fileUploads/MissingFileOverview.jsx [hunk 6]
```diff
@@ -37,7 +37,7 @@ import { getConditionalPages } from '../../utilities';
 import SupportingDocsVerification from './supportingDocsVerification';
 import MissingFileList from './MissingFileList';
 
-const mailInfo = (showOpt = true, address, officeName, faxNum) => {
+const mailInfo = (showOpt = true, address, officeName, faxNum, preamble) => {
```
Comment (michaelclement) [inline src/applications/ivc-champva/shared/components/fileUploads/MissingFileOverview.jsx:40]: We now accept the `preamble` argument which can have JSX in it. If it's not present we default to the original copy.

File: src/applications/ivc-champva/shared/components/fileUploads/MissingFileOverview.jsx [hunk 7]
```diff
@@ -92,11 +96,6 @@ const mailInfo = (showOpt = true, address, officeName, faxNum) => {
   );
 };
 
-export const optionalDescription =
-  'These documents help us process this application faster.';
-export const requiredDescription =
-  'We require these documents in order to process this form.';
-
 // Return a boolean if there are any missing uploads where 'required'
```
Comment (michaelclement) [inline src/applications/ivc-champva/shared/components/fileUploads/MissingFileOverview.jsx:99]: These were never imported anywhere, and now we will accept an optional argument that falls back to these values if need be.

File: src/applications/ivc-champva/shared/components/fileUploads/MissingFileOverview.jsx [hunk 8]
```diff
@@ -199,7 +201,10 @@ export default function MissingFileOverview({
   requiredWarningHeading,
   showMail,
   mailingAddress,
+  mailPreamble,
   officeName,
+  requiredDescription,
+  optionalDescription,
   faxNum,
```
Comment (michaelclement) [inline src/applications/ivc-champva/shared/components/fileUploads/MissingFileOverview.jsx:208]: Adding the three new text customization arguments (docstring above has been updated to reflect)

**Review by balexandr — APPROVED** [review]

**Review by cloudmagic80 — APPROVED** [review]

**Review by micahchiang — APPROVED** [review]

**Pull request merged**